tests/test-event-recorder.c

```diff
@@ -275,6 +275,38 @@ test_event_recorder_record_progress_sync_with_floating_key (struct RecorderFixtu
 }
 
 static void
+test_event_recorder_record_progress_with_same_instance_for_key_and_auxiliary_payload (struct RecorderFixture *fixture,
+                                                                                      gconstpointer           unused)
+{
+  GVariant *variant =
+    g_variant_new_string ("Navas clearly deserved the Golden Glove.");
+  g_variant_ref_sink (variant);
+  emtr_event_recorder_record_start (fixture->recorder, MEANINGLESS_EVENT,
+                                    variant, variant);
+  emtr_event_recorder_record_progress (fixture->recorder, MEANINGLESS_EVENT,
+                                       variant, variant);
+  emtr_event_recorder_record_stop (fixture->recorder, MEANINGLESS_EVENT,
+                                   variant, variant);
+  g_variant_unref (variant);
```

Comment:
Yeah, all of these tests are just checking to make sure nothing crashes (and to document the C API). I can't think of anything interesting to assert since we deliberately return `void`. I had these tests sitting around on for a branch for a long time and just stumbled across them while working on this issue.